pkg/util/grpcclient/signing_handler_test.go

```diff
@@ -0,0 +1,58 @@
+package grpcclient
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/weaveworks/common/user"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/cortexproject/cortex/pkg/cortexpb"
+)
+
+func TestUnarySigningHandler(t *testing.T) {
```

Comment:
Should we have more test cases for testing `ErrDifferentSignaturePresent` error condition?

Reply:
i can but i would need to manually force that error. lemme do that.